src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi/iscsi.component.spec.ts

```diff
@@ -39,10 +42,88 @@ describe('IscsiComponent', () => {
     component = fixture.componentInstance;
+    tcmuIscsiService = TestBed.get(TcmuIscsiService);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should refresh without stats available', () => {
+    const tcmuiscsiData = {
+      daemons: [
+        {
+          server_hostname: 'node1.ceph.local',
+          version: 'ceph version 13.2.2 (02899bfda814146b021136e9d8e80eba494e1126) mimic (stable)',
+          optimized_paths: 0,
+          non_optimized_paths: 1
+        },
+        {
+          server_hostname: 'node2.ceph.local',
+          version: 'ceph version 13.2.2 (02899bfda814146b021136e9d8e80eba494e1126) mimic (stable)',
+          optimized_paths: 0,
+          non_optimized_paths: 1
+        }
+      ],
+      images: [
+        {
+          device_id: 'rbd/disk_1',
+          pool_name: 'rbd',
+          name: 'disk_1',
+          id: '11ce6b8b4567',
+          optimized_paths: [],
+          non_optimized_paths: ['node1.ceph.local', 'node2.ceph.local']
+        }
+      ]
+    };
+    spyOn(tcmuIscsiService, 'tcmuiscsi').and.callFake(() => of(tcmuiscsiData));
+    component.refresh();
+    expect(component.images[0].stats).toBeUndefined();
+    expect(component.images[0].stats_history).toBeUndefined();
+  });
+
+  it('should refresh with stats', () => {
+    const stats = { rd: 0.0, wr: 2.8, rd_bytes: 0.0, wr_bytes: 1468006.4 };
+    const statsHistory = {
+      rd: [[1540551220, 0.0], [1540551225, 0.0], [1540551230, 0.0]],
+      wr: [[1540551220, 0.0], [1540551225, 0.0], [1540551230, 0.0]],
+      rd_bytes: [[1540551220, 0.0], [1540551225, 0.0], [1540551230, 0.0]],
+      wr_bytes: [[1540551220, 0.0], [1540551225, 0.0], [1540551230, 0.0]]
+    };
+    const tcmuiscsiData = {
+      daemons: [
+        {
+          server_hostname: 'node1.ceph.local',
+          version: 'ceph version 13.2.2 (02899bfda814146b021136e9d8e80eba494e1126) mimic (stable)',
+          optimized_paths: 1,
+          non_optimized_paths: 0
+        },
+        {
+          server_hostname: 'node2.ceph.local',
+          version: 'ceph version 13.2.2 (02899bfda814146b021136e9d8e80eba494e1126) mimic (stable)',
+          optimized_paths: 0,
+          non_optimized_paths: 1
+        }
+      ],
+      images: [
+        {
+          device_id: 'rbd/disk_1',
+          pool_name: 'rbd',
+          name: 'disk_1',
+          id: '11ce6b8b4567',
+          optimized_paths: ['node1.ceph.local'],
+          non_optimized_paths: ['node2.ceph.local'],
+          optimized_daemon: 'node1.ceph.local',
+          optimized_since: 1540551284.2985396,
+          stats: stats,
+          stats_history: statsHistory
+        }
+      ]
+    };
+    spyOn(tcmuIscsiService, 'tcmuiscsi').and.callFake(() => of(tcmuiscsiData));
+    component.refresh();
+    expect(component.images[0].stats).toBe(stats);
+    expect(component.images[0].stats_history).toBe(statsHistory);
```

Comment:
This will always be true, as the object will be changed through the component. You should write out what you expect here, to be sure the expect result is correct.